Firebase/Utilities/Logger/GULLogger.m

```diff
@@ -0,0 +1,273 @@
+// Copyright 2017 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "Private/GULLogger.h"
+
+#import <GoogleUtilities/GULAppEnvironmentUtil.h>
+#import "Public/GULLoggerLevel.h"
+
+#include <asl.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <sys/sysctl.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+/// Arguments passed on launch.
+NSString *const kGULDisableDebugModeApplicationArgument = @"-GULDebugDisabled";
+NSString *const kGULEnableDebugModeApplicationArgument = @"-GULDebugEnabled";
+NSString *const kGULLoggerForceSDTERRApplicationArgument = @"-GULLoggerForceSTDERR";
+
+/// Key for the debug mode bit in NSUserDefaults.
+NSString *const kGULPersistedDebugModeKey = @"/google/utilities/debug_mode";
```

Comment:
This implementation is now split-brained. Why are we handling command-line arguments for debug mode in FIRLogger, but NSUserDefaults here?

Either insulate GULLogger from knowing about this gunk, or just put it all in here. This combined approach is even harder to reason about than the old system.